openapi-contracts/src/main/com/sinch/sdk/domains/sms/adapters/api/v1/BatchesApi.java

```diff
@@ -115,10 +115,9 @@ private HttpRequest cancelBatchMessageRequestBuilder(String servicePlanId, Strin
         "/xms/v1/{service_plan_id}/batches/{batch_id}"
             .replaceAll(
                 "\\{" + "service_plan_id" + "\\}",
-                URLParameterUtils.encodeParameterValue(servicePlanId.toString()))
+                URLPathUtils.encodePathSegment(servicePlanId.toString()))
```

Comment:
Nice catch! Hopefully, the service plan id doesn't contain characters that would be encoded differently being a query param or a path param.